caas/clientconfig/k8s.go

```diff
@@ -0,0 +1,131 @@
+package caas
+
+import (
+	"os"
+	"path/filepath"
+
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/cloud"
+)
+
+var logger = loggo.GetLogger("juju.caas.clientconfig")
+
+type K8SClientConfigReader struct {
```

Comment:
I wonder if GetClientConfig should just be a standalone function since the struct doesn't have any data or other methods.